### PR TITLE
redirects from out of process access will go to home page

### PR DESCRIPTION
**Pull request by ptouchton**
## Description
Change the re-directed landing page to the vaos home page if the veteran enters the flow out of sequence.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#34233

## Testing done
Manual, unit and e2e

## Screenshots
New Appt Flow Step 1:
newappt

Step 2:
Landing

Covid Flow Step 1:
covidUrl

Step 2:
Landing

## Acceptance criteria
- [ ] Given any user
When user directly enters any page in the flows (/health-care/schedule-view-va-appointments/appointments/new-appointment* or /health-care/schedule-view-va-appointments/appointments/new-covid-19-vaccine-appointment*)
Then redirect user to the VAOS homepage (/health-care/schedule-view-va-appointments/appointments/)

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by jenniemc — COMMENTED** [review]
The updates certain redirects users to the landing page instead of the start of new appointment flow whenever they refresh or tries to bypass the flow by typing directly in the URL. 

I'm wondering about situation when user book marked start of new appointment flow or type "/new-appointment" directly in the URL. Should that situation also redirect user to landing page? 

I'm posting the question on the original GH tix so we cover all bases.

**Review by vbahinwillit — COMMENTED** [review]
Not sure how to test this.

**Review by jenniemc — CHANGES_REQUESTED** [review]
I'm able to get the TypeOfCarePage.unit.spec.jsx to pass a little further acknowledging there exists the text "primary care" by adding the attribute `newAppointment: { isNewAppointmentStarted: true }`, on line 42 after the user object in the initialState

However, it doesn't seem to acknowledge a history.push.call after the fireEvent.click on /Continue/

File: src/applications/vaos/redux/reducer.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import appointments from '../appointment-list/redux/reducer';
+import newAppointmentReducer from '../new-appointment/redux/reducer';
 
 export default {
-  // newAppointment state slice lazy loaded (new-appointment/redux/reducer.js)
   // expressCare state slice lazy loaded (express-care/redux/reducer.js)
   appointments,
+  newAppointment: newAppointmentReducer,
```
Comment (jenniemc) [inline src/applications/vaos/redux/reducer.js:7]: I'm curious how does adding `newAppointment: newAppointmentReducer` in this reducer file affect the out-of-flow redirect?

File: src/applications/vaos/tests/e2e/community-care.cypress.spec.js [hunk 2]
```diff
@@ -10,13 +10,23 @@ import facilityData from '../../services/mocks/var/facility_data.json';
...
       'health-care/schedule-view-va-appointments/appointments/new-appointment/',
     );
+
+    // we should re-direct to home page
+    cy.get('h2', { timeout: Timeouts.slow })
+      .should('be.visible')
+      .and('contain', 'Your appointments');
+
+    // Start flow
+    cy.findByText('Start scheduling').click();
```
Comment (jenniemc) [inline src/applications/vaos/tests/e2e/community-care.cypress.spec.js:28]: You have cypress working on staging!

**Review by jenniemc — APPROVED** [review]
LGTM!

Just minor suggestions about the new react rules and linting rules. 

The VSP FrontEnd Support recently changed react rules and that's why we are getting warning messages of the DIFF views. Wherever you see "props should not use arrow functions" refer to the [AppointmentListNavigation](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/appointment-list/components/AppointmentListNavigation.jsx#L51-L57) page as an example of how we addressed the warning message.

Another change is linting rule. Wherever you see "use object destructing", hover over the yellow squiggle lines and choose the `quick fix` then `fix this prefer de-constructing problem`

And lastly there is the missing props type validation. I don't see any missing prop validation message in this PR but you will run across it elsewhere in our codebase.

File: src/applications/vaos/redux/reducer.js (hunk 1 above)
Comment (ptouchton) [reply src/applications/vaos/redux/reducer.js:7]: Yeah this took me a few to figure out...Do you remember yesterday when I showed you the "Start schedule" button click event where we where dispatching the "start new appointment" action and then routing to the new-appointment component. What I found was....there was no newAppointment object in the store for the dispatch to change. That object gets initialized when we route to the new-appointment component. By putting this in the base reducer we now have an initialized newApointment for the dispatch to work. Here is the button click event
https://github.com/department-of-veterans-affairs/vets-website/blob/88875c94508f234c077dbca5c62916893772e447/src/applications/vaos/appointment-list/components/ScheduleNewAppointment.jsx#L12-L20
 

And here is where the newApointment object gets added to the store in the route
https://github.com/department-of-veterans-affairs/vets-website/blob/88875c94508f234c077dbca5c62916893772e447/src/applications/vaos/routes.jsx#L36-L43

I think we never really had to rely on that sequence until now....

Comment (jenniemc) [reply src/applications/vaos/redux/reducer.js:7]: Oh yes, I remember now. Great catch!

**Pull request merged**